tsl/src/bgw_policy/job_api.c

```diff
@@ -0,0 +1,233 @@
+	BgwJob *job = ts_bgw_job_find(job_id, CurrentMemoryContext, true);
+	Oid owner = get_role_oid(NameStr(job->fd.owner), false);
+
+	if (!has_privs_of_role(GetUserId(), owner))
+		ereport(ERROR,
+				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+				 errmsg("insufficient permissions to delete job for user \"%s\"",
+						NameStr(job->fd.owner))));
+
+	ts_bgw_job_delete_by_id(job_id);
+
+	PG_RETURN_VOID();
+}
+
+/*
+ * CREATE OR REPLACE PROCEDURE run_job(job_id INTEGER)
+ */
+Datum
+job_run(PG_FUNCTION_ARGS)
+{
+	int32 job_id = PG_GETARG_INT32(0);
+	BgwJob *job = ts_bgw_job_find(job_id, CurrentMemoryContext, true);
+
+	tsl_bgw_policy_job_execute(job);
+
+	PG_RETURN_VOID();
+}
+
+/*
+ * CREATE OR REPLACE FUNCTION alter_job(
+ * 0    job_id INTEGER,
+ * 1    schedule_interval INTERVAL = NULL,
+ * 2    max_runtime INTERVAL = NULL,
+ * 3    max_retries INTEGER = NULL,
+ * 4    retry_period INTERVAL = NULL,
+ * 5    scheduled BOOL = NULL,
+ * 6    config JSONB = NULL,
+ * 7    next_start TIMESTAMPTZ = NULL
+ * 8    if_exists BOOL = FALSE,
+ * ) RETURNS TABLE (
+ *      job_id INTEGER,
+ *      schedule_interval INTERVAL,
+ *      max_runtime INTERVAL,
+ *      max_retries INTEGER,
+ *      retry_period INTERVAL,
+ *      scheduled BOOL,
+ *      config JSONB,
+ *      next_start TIMESTAMPTZ
+ * )
+ */
+Datum
+job_alter(PG_FUNCTION_ARGS)
+{
+	BgwJob *job;
+	BgwJobStat *stat;
+	TupleDesc tupdesc;
+	Datum values[ALTER_JOB_NUM_COLS];
+	bool nulls[ALTER_JOB_NUM_COLS] = { false };
+	HeapTuple tuple;
+	TimestampTz next_start;
+
+	int job_id = PG_GETARG_INT32(0);
+	bool if_exists = PG_GETARG_BOOL(8);
+
+	/* First get the job */
+	job = ts_bgw_job_find(job_id, CurrentMemoryContext, false);
+
+	if (!job)
+	{
+		if (if_exists)
+		{
+			ereport(NOTICE,
+					(errmsg("cannot alter policy schedule, policy #%d not found, skipping",
+							job_id)));
+			PG_RETURN_NULL();
+		}
+		else
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
+					 errmsg("cannot alter policy schedule, policy #%d not found", job_id)));
+	}
+
+	ts_bgw_job_permission_check(job);
+
+	if (!PG_ARGISNULL(1))
+		job->fd.schedule_interval = *PG_GETARG_INTERVAL_P(1);
+	if (!PG_ARGISNULL(2))
+		job->fd.max_runtime = *PG_GETARG_INTERVAL_P(2);
+	if (!PG_ARGISNULL(3))
+		job->fd.max_retries = PG_GETARG_INT32(3);
+	if (!PG_ARGISNULL(4))
+		job->fd.retry_period = *PG_GETARG_INTERVAL_P(4);
+	if (!PG_ARGISNULL(5))
+		job->fd.scheduled = PG_GETARG_BOOL(5);
+	if (!PG_ARGISNULL(6))
+		job->fd.config = PG_GETARG_JSONB_P(6);
+
+	ts_bgw_job_update_by_id(job_id, job);
+
+	if (!PG_ARGISNULL(7))
+		ts_bgw_job_stat_upsert_next_start(job_id, PG_GETARG_TIMESTAMPTZ(7));
+
+	/* check caller does accept tuple */
+	if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("function returning record called in context "
+						"that cannot accept type record")));
+
+	stat = ts_bgw_job_stat_find(job_id);
+	if (stat != NULL)
```

Comment:
Nit: Sometimes you compare with `NULL` and sometimes not. I like the below, but know that it's not the case for all.
```suggestion
	if (stat)
```